### PR TITLE
Add some simple secrets documentation

### PR DESCRIPTION
**Pull request by matthchr**

### CODE REVIEW

File: docs/hugo/content/introduction/secrets.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# Handling secrets
+Resources supported by Azure Service Operator may require secrets as input (passwords, SSH keys, etc).
+They may also produce secrets as "output" (storage keys, connection strings, etc). 
+
+ASO has integration with [Kubernetes secrets](https://kubernetes.io/docs/concepts/configuration/secret/) to interact with 
+these different types of secrets.
+
+## How to provide secrets to Azure
+Resources may have fields in their `spec` expect a reference to a Kubernetes `Secret`.
```
Comment (Porges) [inline docs/hugo/content/introduction/secrets.md:9]:
```suggestion
Resources may have fields in their `spec` that expect a reference to a Kubernetes `Secret`. 
```

File: docs/hugo/content/introduction/secrets.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
+# Handling secrets
+Resources supported by Azure Service Operator may require secrets as input (passwords, SSH keys, etc).
+They may also produce secrets as "output" (storage keys, connection strings, etc). 
+
+ASO has integration with [Kubernetes secrets](https://kubernetes.io/docs/concepts/configuration/secret/) to interact with 
+these different types of secrets.
+
+## How to provide secrets to Azure
+Resources may have fields in their `spec` expect a reference to a Kubernetes `Secret`. 
+For example in order to create a VM, you may want to specify a password to enable access to that VM.
```
Comment (Porges) [inline docs/hugo/content/introduction/secrets.md:10]:
```suggestion
For example, in order to create a VM, you may want to specify an SSH password to enable SSH access to that VM.
```

File: docs/hugo/content/introduction/secrets.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  storage:
+    storageSizeGB: 128
+```
+
+## How to retrieve secrets created by Azure
+
+Some Azure resources produce secrets themselves. ASO supports automatically querying these secrets 
+and storing them in the [SecretDestination](https://pkg.go.dev/github.com/Azure/azure-service-operator/v2/pkg/genruntime#SecretDestination) you specify.
+
+These secrets will be written to the destinations you specify once the resource has successfully been provisioned in Azure.
```
Comment (Porges) [inline docs/hugo/content/introduction/secrets.md:43]:
```suggestion
These secrets will be written to the destination(s) you specify once the resource has successfully been provisioned in Azure.
```

**Review by Porges — APPROVED** [review]

**Review by theunrepentantgeek — APPROVED** [review]
With the changes suggested by @Porges, looks good.

**Pull request merged**